packages/spiderfier/services/managers/spider-manager.ts

```diff
@@ -0,0 +1,175 @@
+import {Injectable, NgZone, SkipSelf} from '@angular/core';
+import {Observable} from 'rxjs/Observable';
+import {OverlappingMarkerSpiderfier} from 'ts-overlapping-marker-spiderfier';
+
+import {MarkerManager} from '../../../core/services/managers/marker-manager';
+import {GoogleMapsAPIWrapper} from '../../../core/services/google-maps-api-wrapper';
+import {AgmMarker} from '../../../core/directives/marker';
+import {AgmMarkerSpider} from '../../directives/marker-spider';
+import {MarkerSpiderInstance, SpiderOptions} from '../google-spider-types';
+
+@Injectable()
+export class SpiderManager extends MarkerManager {
+  private _spiderInstance: Promise<MarkerSpiderInstance>;
+  private _resolver: Function;
+
+  constructor(protected _mapsWrapper: GoogleMapsAPIWrapper, protected _zone: NgZone, @SkipSelf() protected _markerManager: MarkerManager) {
+    super(_mapsWrapper, _zone, _markerManager);
+    this._spiderInstance = new Promise<MarkerSpiderInstance>((resolver) => {
+      this._resolver = resolver;
+    });
+  }
+
+  get instance() {
+    return this._spiderInstance;
+  }
+
+  init(options: SpiderOptions): void {
+    this._mapsWrapper.getNativeMap().then(map => {
+      const spider = new OverlappingMarkerSpiderfier(map as any, options);
+      this._resolver(spider);
+    });
+  }
+
+  addMarker(marker: AgmMarker): void {
+    this._markerManager.addMarker(marker, false);
+
+    const markerPromise = this._markerManager.getNativeMarker(marker);
+
+    this._markers.set(marker, markerPromise);
+
+    Promise.all([
+      this._spiderInstance,
+      markerPromise,
+    ]).then(([spider, nativeMarker]) => {
+      spider.addMarker(nativeMarker as any);
+    });
+  }
+
+  deleteMarker(marker: AgmMarker): Promise<void> {
+    return Promise.all([
+      this._spiderInstance,
+      this._markerManager.getNativeMarker(marker),
+    ]).then(([spider, nativeMarker]) => {
+      if (this._markers.has(marker)) {
+        spider.removeMarker(nativeMarker as any);
+
+        this._markers.delete(marker);
```

Comment:
This happens in the _makerManager.addMarker already

Reply:
See previous comment